src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageInfo.cs

```diff
@@ -8,5 +8,6 @@ internal struct SuppressMessageInfo
         public string Scope;
         public string Target;
         public string MessageId;
+        public AttributeData Attribute;
```

Comment:
Lifetime - It is weakly cached onto a conditional weak table keyed off the compilation. This was done in the original design where we used to apply source suppressions as soon as a diagnostic was reported, and didn't want to recompute it every time a new diagnostic comes in. I recently fixed this to do it as a post pass step when client queries for diagnostics. We can probably remove this cache and just compute it on the fly. I'll open a separate work item to track all your suggestions here.

AttributeData is self contained semantic info of bound attribute and declaring syntax ref, which can be asked for syntax.